Model/Cc.php

```diff
@@ -418,7 +418,7 @@ public function refund(\Magento\Payment\Model\InfoInterface $payment, $amount)
         $request=[];
         $request['OrderId'] = str_replace('-capture', '', $payment->getParentTransactionId());
         $request['MerchantId']= $this->reachHelper->getMerchantId();
-        $request['Amount']= $amount;
+        $request['Amount']= $this->reachCurrency->convertCurrency($payment->getOrder()->getOrderCurrencyCode(), $payment->getCreditmemo()->getGrandTotal());
```

Comment:
Could we make this a bit more readable by assigned some of the values into variables?

```
$currencyCode = $payment->getOrder()->getOrderCurrencyCode();
$grandTotal = $payment->getCreditmemo()->getGrandTotal();
$request['Amount']= $this->reachCurrency->convertCurrency($currencyCode, $grandTotal);
```

Same comment would be applicable to the changes in `Model/Paypal.php`